sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -3233,6 +3233,36 @@ export function doyToInstant(doy: DOY_STRING): Temporal.Instant {
 export function durationToHms(time: Temporal.Duration): HMS_STRING {
   let { days, hours, minutes, seconds, milliseconds } = time;
 
+  while (milliseconds >= 1000) {
+    seconds += 1;
+    milliseconds -= 1000;
+  }
+
+  while (milliseconds <= -1000) {
+    seconds -= 1;
+    milliseconds += 1000;
+  }
+
+  while (seconds >= 60) {
+    minutes += 1;
+    seconds -= 60;
+  }
+
+  while (seconds <= -60) {
+    minutes -= 1;
+    seconds += 60;
+  }
+
+  while (minutes >= 60) {
+    hours += 1;
+    minutes -= 60;
+  }
+
+  while (minutes <= -60) {
+    hours -= 1;
+    minutes += 60;
+  }
+
```

Comment:
Instead of looping have you thought about doing something like this. I did something similar with the TS custom working before reworking it to use the time library written for the planning side.  https://github.com/NASA-AMMOS/aerie-ui/pull/820/files#diff-005afc3206d16625a1a62c09f8ba9d96cbe846da3a31c9b3c0b3822fad77cc0cR367-R389

`
  seconds += Math.floor(milliseconds / 1000);
  milliseconds %= 1000;

  minutes += Math.floor(seconds / 60);
  seconds %= 60;

  hours += Math.floor(minutes / 60);
  minutes %= 60;

  days += Math.floor(hours / 24);
  hours %= 24;
  `
You should probably do something if days are over 365 as years are not supported in a hms durations format

Reply:
Unfortunately this alternative algorithm doesn't seem to work quite right for negative values

```
  "metadata": Object {
            "simulatedActivityId": 2028,
          },
          "stem": "PREPARE_LOAF",
          "time": Object {
    -       "tag": "-001T12:16:54.000",
    +       "tag": "-003T14:18:54.000",
            "type": "EPOCH_RELATIVE",
          },
          "type": "command",
        },
```

Reply:
The code in that time library seems to assume that the following is true:

```
x == y * Math.floor(x / y) + (x % y) 
```

Testing with a positive value:
```
var x = 10.4;
var y = 5;
console.log(y * Math.floor(x / y) + (x % y));

// 10.4
```

Testing with a negative value:
```
var x = -10.4;
var y = 5;
console.log(y * Math.floor(x / y) + (x % y));

// -15.4
```

Reply:
awww yeah I forgot you can have a negative duration.